packages/yew/src/html/classes.rs

```diff
@@ -16,8 +15,28 @@ pub struct Classes {
     set: IndexSet<Cow<'static, str>>,
 }
 
+/// helper method to efficiently turn a set of classes into a space-separated
+/// string. Abstracts differences between ToString and IntoPropValue. The
+/// `rest` iterator is cloned to pre-compute the length of the String; it
+/// should be cheap to clone.
+fn build_string<'a>(first: &'a str, rest: impl Iterator<Item = &'a str> + Clone) -> String {
+    // The length of the string is known to be the length of all the
+    // components, plus one space for each element in `rest`.
+    let mut s = String::with_capacity(
+        rest.clone()
+            .map(|class| class.len())
+            .chain([first.len(), rest.size_hint().0])
+            .sum(),
```

Comment:
Sure, will add comments. I actually ended up with this implementation because of how `sum` doesn't do well with type inference when participating in arithmetic, but yeah comments will definitely clear it up.

Reply:
Done